kafka-connect-influxdb/src/test/scala/com/datamountaineer/streamreactor/connect/influx/InfluxBatchPointsBuilderFnTest.scala

```diff
@@ -0,0 +1,757 @@
+    PointMapFieldGetter.measurement(point) shouldBe measurement
+    val time = PointMapFieldGetter.time(point)
+    before <= time shouldBe true
+    time <= System.currentTimeMillis() shouldBe true
+
+    val map = PointMapFieldGetter.fields(point)
+    map.size shouldBe 14
+
+    map.get("_id") shouldBe "580151bca6f3a2f0577baaac"
+    map.get("index") shouldBe 0
+    map.get("guid") shouldBe "6f4dbd32-d325-4eb7-87f9-2e7fa6701cba"
+    map.get("isActive") shouldBe false
+    map.get("balance") shouldBe 3589.15
+    map.get("age") shouldBe 27
+    map.get("eyeColor") shouldBe "brown"
+    map.get("this_is_renamed") shouldBe "Clements Crane"
+    map.get("company") shouldBe "TERRAGEN"
+    map.get("email") shouldBe "clements.crane@terragen.io"
+    map.get("phone") shouldBe "+1 (905) 514-3719"
+    map.get("address") shouldBe "316 Hoyt Street, Welda, Puerto Rico, 1474"
+    map.get("latitude") shouldBe "-49.817964"
+    map.get("longitude") shouldBe "-141.645812"
+  }
+
+  "convert a sink record with a json string payload with specific fields being selected" in {
+    val jsonPayload =
+      """
+        | {
+        |    "_id": "580151bca6f3a2f0577baaac",
+        |    "index": 0,
+        |    "guid": "6f4dbd32-d325-4eb7-87f9-2e7fa6701cba",
+        |    "isActive": false,
+        |    "balance": 3589.15,
+        |    "age": 27,
+        |    "eyeColor": "brown",
+        |    "name": "Clements Crane",
+        |    "company": "TERRAGEN",
+        |    "email": "clements.crane@terragen.io",
+        |    "phone": "+1 (905) 514-3719",
+        |    "address": "316 Hoyt Street, Welda, Puerto Rico, 1474",
+        |    "latitude": "-49.817964",
+        |    "longitude": "-141.645812"
+        | }
+      """.stripMargin
+
+    val topic = "topic1"
+    val measurement = "measurement1"
+
+    val before = System.currentTimeMillis()
+
+    val record = new SinkRecord(topic, 0, null, null, Schema.STRING_SCHEMA, jsonPayload, 0)
+
+    val extractor = StructFieldsExtractor(false, Map("_id" -> "_id", "name" -> "this_is_renamed", "email" -> "email"), None, Set.empty)
+    val settings = InfluxSettings("connection", "user", "password", "database1", Map(topic -> measurement), Map(topic -> extractor))
+    val batchPoints = InfluxBatchPointsBuilderFn(Seq(record), settings)
+    val points = batchPoints.getPoints
+    points.size() shouldBe 1
+    val point = points.get(0)
+    PointMapFieldGetter.measurement(point) shouldBe measurement
+    val time = PointMapFieldGetter.time(point)
+    before <= time shouldBe true
+    time <= System.currentTimeMillis() shouldBe true
+
+    val map = PointMapFieldGetter.fields(point)
+    map.size shouldBe 3
+
+    map.get("_id") shouldBe "580151bca6f3a2f0577baaac"
+    map.get("this_is_renamed") shouldBe "Clements Crane"
+    map.get("email") shouldBe "clements.crane@terragen.io"
+
+  }
+
+  "throw an error of if nested json since there is no flattening of json for a sink record with string json payload" in {
```

Comment:
I would object to that. We should provide a mechanism like

```flattenJson = '.'```

In that case we would automatically flatten the json structure using this separator

i.e. 

```  """
         | {
         |    "eyeColor": "brown",
         |    "name.first": "Christian",
         |    "name.last": "Melton"
         |    }
         | }```